contract/security/contract_security.xml

```diff
@@ -7,6 +7,13 @@
         <field name="domain_force">['|', ('company_id', '=', False), ('company_id', 'child_of', [user.company_id.id])]</field>
     </record>
 
+    <record id="rule_contract_contract_portal" model="ir.rule">
+        <field name="name">Contract contract portal</field>
+        <field name="model_id" ref="model_contract_contract"/>
+        <field name="domain_force">[('message_partner_ids', 'child_of', [user.partner_id.commercial_partner_id.id])]</field>
```

Comment:
I think nobody will want that all the contacts of a company see these contracts. We should put a more stricter one, only having visibility to direct followers.